app/main.py

```
@@ -36,89 +126,80 @@ def process_csv(file_content: bytes, filename: str) -> None:
-def read_root() -> BaseResponse:
-    """
-    Root endpoint that returns a welcome message.
-    
-    Returns:
-        BaseResponse: A response indicating success with a welcome message.
-    """
-    return BaseResponse(success=True, message="Welcome to the User Management API")
+        processing_status[file_id]["status"] = "failed"
+        processing_status[file_id]["error"] = str(e)
+        logging.error(f"Error processing CSV '{filename}': {str(e)}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)